crates/storage/store/storage.rs

```diff
@@ -210,51 +218,87 @@ impl Store {
     }
 
     /// Add transaction to the pool
-    pub fn add_transaction_to_pool(
-        &self,
-        hash: H256,
-        transaction: MempoolTransaction,
-    ) -> Result<(), StoreError> {
-        self.engine.add_transaction_to_pool(hash, transaction)
-    }
-
-    /// Get a transaction from the pool
-    pub fn get_transaction_from_pool(
-        &self,
-        hash: H256,
-    ) -> Result<Option<MempoolTransaction>, StoreError> {
-        self.engine.get_transaction_from_pool(hash)
+    pub fn add_transaction_to_pool(&self, transaction: MempoolTransaction) {
+        let mut mempool = self.mempool.lock().unwrap();
+        let maybe_old_value = mempool.get_mut(&transaction.sender());
+        let sender = transaction.sender();
+
+        match maybe_old_value {
+            Some(old_value) => {
+                old_value.insert(transaction.nonce(), transaction);
+            }
+            None => {
+                let mut new_value = BTreeMap::new();
+                new_value.insert(transaction.nonce(), transaction);
+                mempool.insert(sender, new_value);
+            }
```

Comment:
Could we replace this match with `entry(..).or_default().insert(..)`?

Reply:
Great suggestion!